src/pylife/materiallaws/notch_approximation_law.py

```diff
@@ -674,140 +581,458 @@ def stress_secondary_branch(self, delta_load, *, rtol=1e-5, tol=1e-6):
-            # get the class index of the corresponding bin/class
-            class_index = lut_for_first_node.searchsorted(first_abs_load)
-        
-            max_class_index = max(self._lut_secondary_branch.index.get_level_values("class_index"))
-                
-            # raise error if requested load is higher than initialized maximum absolute load
-            if class_index+1 > max_class_index:
-                raise ValueError(f"Binned class is initialized with a maximum absolute delta_load load of {2*self._maximum_absolute_load}, "\
-                                 f" but a higher absolute delta_load value of {first_abs_load} is requested (in stress_secondary_branch()).")
-        
-            # sign is a DataFrame with one column, convert to series
-            sign = sign[sign.columns[0]]
+    def _stress_multiple_assessment_points_single_lut(self, load):
+        '''The stress of the primary path in the stress-strain diagram at a given load
+        by using the value of the look-up table.
+        
+        This method performs the task for for multiple points at once, 
+        i.e. delta_load is a DataFrame with values for every node.
+
+        .. note::
             
-            # get stress from matching class, "+1", because the next higher class is used
-            delta_stress = self._lut_secondary_branch[self._lut_secondary_branch.index.get_level_values("class_index") == class_index+1].delta_stress 
+            The exact value would be computed by ``self._notch_approximation_law.stress(load)``.
         
-            # multiply with sign
-            return sign * delta_stress.reset_index(drop=True)
-             
-        # if the assessment is performed for multiple points at once, but only one lookup-table is used
-        elif isinstance(delta_load, pd.DataFrame):
+        Parameters
+        ----------
+        load : array-like float
+            The load, either a scalar value or a pandas DataFrame with RangeIndex (no named index)
+
+        Returns
+        -------
+        stress : array-like float
+            The resulting stress
+        '''
+        sign = np.sign(load)
         
-            delta_load = delta_load.fillna(0)
-            sign = sign.fillna(0)
-            
-            index = self._lut_secondary_branch.delta_load.searchsorted(np.abs(delta_load.values))-1   # "-1", transform to zero-based indices
+        load = load.fillna(0)
+        sign = sign.fillna(0)
+        
+        index = self._lut_primary_branch.load.searchsorted(np.abs(load.values))-1   # "-1", transform to zero-based indices
+
+        # raise error if requested load is higher than initialized maximum absolute load
+        if np.any(index+1 >= len(self._lut_primary_branch)):
+            raise ValueError(f"Binned class is initialized with a maximum absolute load of {self._maximum_absolute_load}, "\
+                                f" but a higher absolute load value of |{load.max()}| is requested (in stress()).")
```

Comment:
Maybe we can `._find_load_index_or_raise(load)` which finds the `index` of the load or raises if any of them is outside. Would remove code duplication and help readability.

Or at least introduce a method `._raise_if_requested_load_higher_than_max_abs_load(index, max_abs_load, max_load)`